cmake/pcl_find_avx.cmake

```diff
@@ -0,0 +1,39 @@
+###############################################################################
+# Check for the presence of AVX and figure out the flags to use for it.
+macro(PCL_CHECK_FOR_AVX)
+  set(AVX_FLAGS)
+  set(AVX_DEFINITIONS)
+
+  include(CheckCXXSourceRuns)
+
+  check_cxx_source_runs("
+    #include <immintrin.h>
+    int main()
+    {
+      __m256 a;
+      a = _mm256_set1_ps(0);
+      return 0;
+    }"
+  HAVE_AVX)
+  
+  check_cxx_source_runs("    
+    #include <immintrin.h>
+    int main()
+    {
+      __m256i a = {0};
+      a = _mm256_abs_epi16(a);
+      return 0;
+    }"
+  HAVE_AVX2)
+  
+  if(MSVC)
+    if(HAVE_AVX)
+      string(APPEND AVX_DEFINITIONS " -D__AVX__")
+      set(AVX_FLAGS "/arch:AVX")
+    endif()
+    if(HAVE_AVX2)
+      string(APPEND AVX_DEFINITIONS " -D__AVX2__")
+      set(AVX_FLAGS "/arch:AVX2")
```

Comment:
I checked the [documentation](https://docs.microsoft.com/en-us/cpp/build/reference/arch-x64): The definition is not required.

Another point: You could check first for AVX2 and when the check succeed skip the AVX test to speedup the CMake script.

Reply:
Ahh, yes it does set them when just using the /arch definition. I'll updated with your suggestions.